astropy/io/votable/converters.py

```diff
@@ -674,19 +674,25 @@ def __init__(self, field, config=None, pos=None):
 
         precision = field.precision
         width = field.width
-        format_parts = ['%']
+
+        if precision is None:
+            format_parts = ['{0']
```

Comment:
Finally found the solution, apparently ``!r`` works totally differently that I assumed. Sometimes it makes sense to actually read the manual 😊